x-pack/plugins/fleet/common/services/validate_package_policy.ts

```diff
@@ -143,7 +143,7 @@ export const validatePackagePolicy = (
         results[name] = input.enabled
           ? validatePackagePolicyConfig(
               configEntry,
-              inputVarDefsByPolicyTemplateAndType[inputKey][name],
+              (inputVarDefsByPolicyTemplateAndType[inputKey] ?? {})[name],
```

Comment:
Should we have a second level of inputVarDefs just by input type if there is no `policy_template` something like this? otherwise we loose validation and I do not think we should allow to create inputs that are not valid.  

```
inputVarDefsByPolicyTemplateAndType[inputKey] 
  ? inputVarDefsByPolicyTemplateAndType[inputKey][name] 
  :  (inputVarDefsByType[inputType] ?? {})[name] // here I am wondering if we should not throw if there is no input matching the input type
```

Reply:
I'm not sure what do you mean to put in `inputVarDefsByType`, in this scenario, there are empty vars in `policy_templates.inputs.linux/metrics.vars`. The validation is already there for `data_streams.streams.input.vars`.

See the package details:
```
{
    "item": {
        "policy_templates": [
            {
                "name": "system",
                "title": "Linux kernel metrics",
                "description": "Collect system metrics from Linux operating systems",
                "inputs": [
                    {
                        "title": "Collect low-level system metrics from Linux instances",
                        "vars": [],
                        "type": "linux/metrics",
                        "description": "Collecting Linux conntrack, ksm, pageinfo metrics."
                    }
                ],
                "multiple": true
            }
        ],
        "data_streams": [
            {
                "dataset": "linux.memory",
                "package": "linux",
                "path": "memory",
                "streams": [
                    {
                        "input": "linux/metrics",
                        "title": "Linux memory metrics",
                        "vars": [
                            {
                                "name": "period",
                                "type": "text",
                                "title": "Period",
                                "multi": false,
                                "required": true,
                                "show_user": true,
                                "default": "10s"
                            }
                        ],
                        "template_path": "stream.yml.hbs",
                        "description": "Linux paging and memory management metrics"
                    }
                ],
                "title": "Linux-only memory metrics",
                "release": "experimental",
                "type": "metrics"
            }
        ]
    }
}
```

Reply:
Here the `validatePackagePolicyConfig` function take as the first argument the user provided value and as the second one the package provided spec to validate against if we do not provide a second argument we are not validating the input. 
What I suggest is in case we are not able to map the user provided input to a package var def to try to map only using the input type.

Does it make clearer?

Reply:
I see what you mean, though in this scenario, `inputVarDefsByType` would build the same object as `inputVarDefsByPolicyTemplateAndType` (none of the policy_templates have that var provided by the user, the vars are really just misplaced in the request):
`hasIntegration` is false here, so the object keys will be` input.type `only:
https://github.com/elastic/kibana/blob/main/x-pack/plugins/fleet/common/services/validate_package_policy.ts#L104

the question is, what should happen if there are extra vars provided that are not there in the template, should there be a validation error coming?

Reply:
I agree with your assessment. I'll update the pr to give a validation error instead of silencing the error in case of extra/unexpected vars. I think this change is not breaking in a sense that it only improves the error message (the request would fail with undefined error anyway). 
Though the original [change](https://github.com/elastic/kibana/pull/123153) might cause some unexpected impact where this API was called with invalid package request, like in e2e-testing.